indexing-service/src/main/java/org/apache/druid/indexing/overlord/ForkingTaskRunner.java

```diff
@@ -219,6 +219,24 @@ public TaskStatus call()
                           );
                         }
 
+                        // Override task specific javaOptsArray
+                        Object taskJavaOptsArray = task.getContextValue(
+                            ForkingTaskRunnerConfig.JAVA_OPTS_ARRAY_PROPERTY
+                        );
+                        if (taskJavaOptsArray != null) {
+                          if (taskJavaOptsArray instanceof List) {
```

Comment:
instead of this - can we try `List<String> taskJavaOptsArray = jsonMapper.convertValue(task.getContextValue(ForkingTaskRunnerConfig.JAVA_OPTS_ARRAY_PROPERTY), new TypeReference<List<String>>() {});` and then add that whole array to the command? Also, I think that we should throw an error incase of a bad format instead of silently ignoring the overrides. 

Please write a UT where the override (and hence setting) can be tested between the deprecated prop and the new prop

Reply:
done